### DISCUSSION TITLE
A "just make it build" button

### DISCUSSION THREAD

**Discussion by daiplusplus** [dotnet/roslyn#72698] [Ideas]
* At any given point in time, whatever C# codebase I'm working on, is probably in some broken, unbuildable state, as I can never seem to resist adding major-refactoring work to every expected 1-line change <sub>😬</sub>.
* But I also often need to be able to get the project back into a buildable state at short-notice, often for things like being able to run unit-tests, even if the main program-flow is horribly broken due to runtime bugs, but at least it builds.
*  Or consider a typical _whoopsie_ situation where an entire 100+ project Enterprise(TM) software solution - comprised of 99.99% _perfect_, bug-free code, hours away from deployment - is stuck waiting on the tarmac because a single tiiiiny, but commonly referenced dependency, can't build because of a newly-introduced syntax error, or compiler complaint about a missing member or function or nonexistent parameter - normal _post-refactoring fun_ stuff. That something so small can ground something so big is more galling when the compiler errors are in some part of the codebase that would never even be invoked normally.
    * During times like these, it's easy to get overwhelmed by having to suddenly drop-everything to figure out why it's broken (context-switching is expensive!) just so you can get the build on track so you can quickly run those unit and integration tests before breaking the codebase again for the next refactoring step - and then forgetting what it was you were originally trying to accomplish anyway.

In situations like these, especially if it happens at 3am, the way out is to just _mechanically_ address each syntax error reported by the compiler, while the "interface not implemented" and "function does not return a value" errors can be resolved with a swift application of copy+pasting "`throw new NotImplementedException();`" a few dozen times throughout your codebase. There. _Boom_. Now it builds! The unit tests pass. Hurrah!

After having that happen to me far-too-often now - and rather than doing the _sensible thing_ and try to change my working habits - I realise that what I want is a big fat (and ideally red-coloured) button labelled "make it build". This button wouldn't be magic, nor would it need to pull-in any AI/LLM/ChatGPT services either: it would simply and _mechanically_ replace every compiler-reported syntax-error with an `#if TODO_FIX_LATER ... #endif` which would surround whatever syntactically valid braces, statements/control-structures indentation remain - thus fixing any syntax or structural errors in the compiler, while also strategically placing those `throw new NotImplementedException()` landmines (and more `#if` regions) at-and-around applicable compiler errors. It's  what I already do, but doing this manually is very tedious (doubly-so because eventually we need to remove those changes) which just isn't a good use of our time. Because it's such a simple process it's something that I believe the IDE tooling (e.g. a Roslyn-based analyzer, with the ability to apply patches/edits to code) could do for us.

Assuming this would be an effective way to save dev-time, while being as straighttforward and simple to implement as my sheer naïveté in this problem-space then it probably is worth it for me to invest some time in building it using Roslyn-based tooling, perhaps as a VS VSIX Extension (for the ability to apply edits to code, something Roslyn analyzers can't do by themselves - and for getting to create that lovely, lovely big red button I keep fantasizing about. Maybe it will get really popular enough that MS builds it right into VS...

I'd like to know if this idea has been attempted before - and/or if this something no-one is really itching for and I need better hobbies.

I did some cursory research and found some interesting things like [this MSR paper from 2012 about _automatic program repair_](https://www.microsoft.com/en-us/research/wp-content/uploads/2012/10/res0099-logozzo.pdf), though they're looking at automatically resolving correctness bugs in code that already compiles, whereas my interest is only about sweeping compiler errors under-the-rug. I also I assume my searches are returning little because I'm probably using the wrong terminology/search-terms - so please speak-up if there's something I've missed - or if you actually think this might be a good idea? (_hah_)

**Comment by KennethHoff** [comment]
Being able to say "I don't care" to Roslyn Analyzers at the very least would be useful; compiler-level errors (say an invalid class name) I could see being much more complicated to implement

**Reply by alexbdaniel** [reply]
It could help attract new people to c#. Being able to run things quickly is partly (imo) why JS is popular. And
